File: desert/sarven-sentry.lua
function pos2xy(pos) return pos['x'], pos['y'] end

loop
    local flagGreen = self:findFlag("green")
    local flagBlack = self:findFlag("black")
    local flagViolet = self:findFlag("violet")

    if flagGreen then
        self:pickUpFlag(flagGreen)
        self:buildXY("fence", pos2xy(flagGreen.pos))
    elseif flagBlack then
        self:pickUpFlag(flagBlack)
        self:buildXY("fire-trap", pos2xy(flagBlack.pos))
    elseif flagViolet then
        self:pickUpFlag(flagViolet)
    end
end

File: desert/sarven-sentry.clj
(while true
  (let [flagGreen (.findFlag this "green")
        flagBlack (.findFlag this "black")
        flagViolet (.findFlag this "violet")]
    (if flagGreen
      (when true
        (.pickUpFlag this flagGreen)
        (.buildXY this "fence" (.x (.pos flagGreen)) (.y (.pos flagGreen))))
      (if flagBlack
        (when true
          (.pickUpFlag this flagBlack)
          (.buildXY this "fire-trap" (.x (.pos flagBlack)) (.y (.pos flagBlack))))
        (if flagViolet
          (.pickUpFlag this flagViolet)
          (.wait this 0.1))))))
